### PR TITLE
Task 61 Refresh Token

### PR DESCRIPTION
**Pull request by melonholy**

### CODE REVIEW

File: api/middlewares/refreshAccessToken.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+const refreshToken = (req, res, next) => {
+  let decodedToken = jwt.decode(req.headers.token);
   const user = {
-    id: decodedToken.sub,
-    name: decodedToken.name,
+    id: decodedToken.id,
+    name: decodedToken.email,
     expiresIn: process.env.EXPIRES_IN,
   };
   const token = jwt.sign(user, process.env.SECRET);
```
Comment (tresdosdos) [inline api/middlewares/refreshAccessToken.js]: jwt.sign has third argument = options, one of them is expiresIn

File: api/middlewares/refreshAccessToken.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-  req.headers.authorization = token;
+  req.headers.token = token;
```
Comment (tresdosdos) [inline api/middlewares/refreshAccessToken.js]: set it to Authorization header

File: api/routes/auth.js [hunk 3]
```diff
@@ -208,6 +208,10 @@ router.post('/reset/:linkId', controllers.resetApprovementPassword);
 
 router.post('/reset-password', controllers.resetPassword);
 
-router.put('/change-password/:userId', controllers.changePassword);
+router.put(
+  '/change-password/:userId',
+  refreshToken,
```
Comment (tresdosdos) [inline api/routes/auth.js:215]: you can write it before any of route with imperative writing, check how middleware is working at express.

**Review by tresdosdos — CHANGES_REQUESTED** [review]

File: api/routes/auth.js (hunk 3 above)
Comment (OlegPanasyuk) [reply api/routes/auth.js]: Do you mean that it possible to use it like `app.use(refreshToken)`?

Comment (OlegPanasyuk) [reply api/routes/auth.js:215]: > you can write it before any of route with imperative writing, check how middleware is working at express.

Do you mean that it possible to use it like app.use(refreshToken)?

**Review by OlegPanasyuk — APPROVED** [review]

**Review by sergevei — APPROVED** [review]

**Pull request merged**